File: PRACTICA#3/PERSISTENCIA/3/PYTHON/ArchivoCliente.py
from Cliente import Cliente

class ArchivoCliente:
    def __init__(self, nomA: str):
        self.nomA = nomA
    
    def crearArchivo(self):
        try:
            with open(self.nomA, 'w') as f:
                f.write("")
        except IOError as e:
            print(f"Error creando archivo: {e}")
    
    def guardarCliente(self, c: 'Cliente'):
        try:
            with open(self.nomA, 'a') as f:
                f.write(f"{c.id},{c.nombre},{c.telefono}\n")
        except IOError as e:
            print(f"Error guardando cliente: {e}")
    
    def buscarCliente(self, c: int):
        try:
            with open(self.nomA, 'r') as f:
                for line in f:
                    id, nombre, telefono = line.strip().split(',')
                    if int(id) == c:
                        return Cliente(int(id), nombre, int(telefono))
        except IOError as e:
            print(f"Error buscando cliente: {e}")
        return None
    
    def buscarCelularCliente(self, c: int):
        try:
            with open(self.nomA, 'r') as f:
                for line in f:
                    id, nombre, telefono = line.strip().split(',')
                    if int(id) == c:
                        return Cliente(int(id), nombre, int(telefono))
        except IOError as e:
            print(f"Error buscando celular: {e}")
        return None

File: PRACTICA#3/PERSISTENCIA/3/JAVA/ArchivoCliente.java
import java.io.*;

public class ArchivoCliente {
    private String nomA;

    public ArchivoCliente(String nomA) {
        this.nomA = nomA;
    }

    public void crearArchivo() {
        try (FileWriter f = new FileWriter(nomA)) {
        } catch (IOException e) {
            System.err.println("Error creando archivo: " + e.getMessage());
        }
    }

    public void guardarCliente(Cliente c) {
        try (FileWriter f = new FileWriter(nomA, true)) {
            f.write(c.getId() + "," + c.getNombre() + "," + c.getTelefono() + "\n");
        } catch (IOException ex) {
            System.err.println("Error guardando cliente: " + ex.getMessage());
        }
    }

    public Cliente buscarCliente(int c) {
        try (BufferedReader f = new BufferedReader(new FileReader(nomA))) {
            String line;
            while ((line = f.readLine()) != null) {
                String[] parts = line.split(",");
                if (Integer.parseInt(parts[0]) == c) {
                    return new Cliente(Integer.parseInt(parts[0]), parts[1], Integer.parseInt(parts[2]));
                }
            }
        } catch (IOException e) {
            System.err.println("Error buscando cliente: " + e.getMessage());
        }
        return null;
    }

    public Cliente buscarCelularCliente(int c) {
        try (BufferedReader f = new BufferedReader(new FileReader(nomA))) {
            String line;
            while ((line = f.readLine()) != null) {
                String[] parts = line.split(",");
                if (Integer.parseInt(parts[0]) == c) {
                    return new Cliente(Integer.parseInt(parts[0]), parts[1], Integer.parseInt(parts[2]));
                }
            }
        } catch (IOException e) {
            System.err.println("Error buscando celular: " + e.getMessage());
        }
        return null;
    }
}
